### ISSUE TITLE
variable shadowing with destructuring

### ISSUE THREAD

**Issue by aeosynth** [babel/babel#1136]
```js
import {f} from 'module'

class Foo {
  method() {
    let {x} = f()
  }
}
```

```js
...
var _f = require('module');
...
  _createClass(Foo, [{
    key: 'method',
    value: function method() {
      var _f = _f.f();

      var x = _f.x;
    }
  }]);
...
```

`var _f` is used twice

**Comment by sebmck** [comment]
Looks like the uid in `method` is being generated before the module import one. Might need to revert back to a global uid registry since I don't see a way around this.

**Comment by sebmck** [comment]
Should be fixed as of 5.0.3, thanks!
